File: makro_board_client/lib/models/Control.dart
import 'package:json_annotation/json_annotation.dart';

import 'view_config_parameter.dart';
import 'view.dart';

@JsonSerializable()
class Control {
  final String symbolicName;
  final View view;
  final List<ViewConfigParameter> configParameters;
  final List<Control>? subControls;

  Control({
    required this.symbolicName,
    required this.view,
    required this.configParameters,
    required this.subControls,
  });

  Control.fromJson(Map<String, dynamic> json)
      : symbolicName = json["symbolicName"],
        view = View.fromJson(json["view"]),
        configParameters = List.castFrom(json["configParameters"]).map<ViewConfigParameter>((jsonConfigParameter) => ViewConfigParameter.fromJson(jsonConfigParameter)).toList(),
        subControls = json["subControls"] == null ? null : List.castFrom(json["subControls"]).map<Control>((jsonControl) => Control.fromJson(jsonControl)).toList();
}

File: MakroBoard/ApiModels/Control.cs
using System.Collections.Generic;

namespace MakroBoard.ApiModels
{
    public class Control
    {
        public Control(string symbolicName, View view, ConfigParameters configParameters, IEnumerable<Control> subControls)
        {
            SymbolicName = symbolicName;
            View = view;
            ConfigParameters = configParameters;
            SubControls = subControls;
        }

        public string SymbolicName { get; }

        public View View { get; }
        public ConfigParameters ConfigParameters { get; }

        public IEnumerable<Control> SubControls { get; }
    }
}
